### ISSUE TITLE
TRACKING: JOSS review response

### ISSUE THREAD

**Issue by lgeiger** [larq/larq#243]
## Full Review: https://github.com/openjournals/joss-reviews/issues/1746

This issue tracks the progress of the response for the JOSS paper review.
I will only touch the comments related to the library and documentation for now.

## Review Comments related to Library and Documentation

- [x] The documentation clearly states that it is designed to solve power consumption and speed issues with deep neural network architectures. A notion of the target audience is mentioned on the 'about' page ("researchers and developers"), 'papers using Larq' page and in the last paragraph of the paper. I suggest clarifying this directly on the homepage. ➡️ I tried to clarify it in https://github.com/larq/larq/pull/261

- [x] The documentation and paper make no claims on the comparison of performance between Quantized Neural Networks and their high-precision weight and bias counterparts. At the time of writing, there is clearly a trade-off (although the authors are working hard to close the gap). I suggest being transparent in this difference and possibly opening the option of educating users on when a few percents of accuracy on a specific dataset are essential and when not. This might be especially relevant for people working in the industry.
   ➡️ I agree. I made an issue to track our progress on this: https://github.com/larq/larq/issues/262

- [x] There is a community menu item with a contribution guide tab in the documentation. This clearly states the steps for third parties wishing to contribute to the software (point 1). The other use cases are undocumented. Links to GitHub issues and spectrum are missing in the document :arrow_right: https://github.com/larq/larq/pull/237

- [x] The authors provide numerous examples. There are even hosted examples. In general, the examples are well-documented, except for the Larq Zoo. There is minimal context at these examples, which are more like code snippets. I suggest extending the description for these examples somewhat. :arrow_right: https://github.com/larq/larq/pull/235

- [x] For each of the items in the API, one could ask: how does this compare to the underlying framework and why does it need an extension? Most of the sections (e.g. Quantizers) answer these questions, for example by providing references. An example where this information is missing is sign. Adding a reference to the tensorflow sign and/or mentioning that tf.math.sign would be a quick win. :arrow_right: https://github.com/larq/larq/pull/238

- [x] The FAQ is now located under "Why Larq?", which I found to be an illogical location. :arrow_right: https://github.com/larq/larq/pull/239

- [x] Inline math is not rendered (ex. key concepts). :arrow_right: https://github.com/larq/larq/pull/228

- [x] Consider adding references to the home page (e.g. Courbariaux). ➡️ https://github.com/larq/larq/pull/258

- [x] Suggestions for spelling and grammar can be found here. :arrow_right: https://github.com/larq/larq/pull/227 Thanks for the fixes :+1:

- [x] Installation instructions: Installation (https://github.com/larq/larq#installation and https://larq.dev/#installation) also needs to also document pip install tensorflow, pip install tensorflow-gpu (or add them to setup.py etc), referencing the earlier link (https://www.tensorflow.org/install) for additional dependencies. :arrow_right: Unfortunately we can't add `tensorflow` as a direct dependency because that would break usage with GPU. We clarified the instructions: https://github.com/larq/larq/pull/240

- [x] Performance: There is frequent mention of the efficiency of BNNs, however it would be nice to have e.g. a quantitative table of results showing that larq matches theoretical/literature speed improvements.
   ➡️ I fully agree, that this would be really useful. However, Larq is currently only focused on research and training extremely quantized networks. That means there is currently no deployment workflow available and no optimized kernels implemented. Quoting theoretical numbers is always tricky since it really depends on the platform and runtime the models are deployed.
That being said, we are working on a compute engine for Larq that will allow optimized deployment using TFLite on ARM and mobile platforms at which point we will publish real world numbers and benchmarks as soon as possible.

- [x] Submission requirements: I'm not sure but as per https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements the software seems to be bordering on a "'thin' API client" around tensorflow.keras. Perhaps it's worth submitting to JOSE instead? Or including in a larger publication similar to the Helwegen et al. paper? For example ls larq/_.py larq/snapshots/_.py | grep -v test | xargs cloc yields 1,586 non-blank/comment lines, with the main layers.py file exclusively made up of lines which look like.
   ➡️ It is true that the API closely mirrors `tf.keras`. However, this was a deliberate design choice early on in order to be compatible with the entire ecosystem and not introduce a new abstraction in order to be easy to adopt for existing Keras users. We explored some different design ideas early on in https://github.com/larq/larq/issues/4 and concluded that the current API design is probably the most ergonomic for users.
[`larq/layers.py`](https://github.com/larq/larq/blob/master/larq/layers.py) is purely for documentation purpose. The (also pretty straight forward) actual implementations are located under [`larq/layers_base.py`](https://github.com/larq/larq/blob/master/larq/layers_base.py). While Larq layers only add a few keyword argument to compared to `tf.keras` layers, under the hood they implement functionality that isn't available otherwise without completely reimplementing the custom layers from scratch. In addition to the layers we have added metrics and quantizers specific to binarized and other extremely quantized neural networks which are too specific to upstream into `tf.keras` itself.

- [x] A statement of need: https://larq.dev/about/ doesn't mention students (while the paper does). :arrow_right: https://github.com/larq/larq/pull/241

- [x] Functionality documentation: The API link (https://larq.dev/api/layers/) seems to imply this is just larq.layers, yet the Models link (https://larq.dev/models/) gives a separate page which seems to be detailing API for larq/zoo which seems to be a separate package built on top of the larq.layers API. If this is the case it needs to be made clearer. :arrow_right: We tried to make the distinction clearer: https://github.com/larq/larq/pull/242

- [x] The documentation lacks formality (in a mathematical sense). I miss a proper definition of key concepts such as "quantizer" and others.
   ➡️ We intentionally opted for a less formal language to reduce jargon and make the documentation a bit more approachable for people without a math background. However I see your point that our explanation is unclear from a mathematical view. I tried to add 
 a small mathematical clarification in https://github.com/larq/larq/pull/259

- [x] In the Home section the first example of a model is shown. I think it should be described with more details. ➡️ I extended the description and added a link to the more detailed user guide: https://github.com/larq/larq/pull/261

- [x] There is a LaTex compilation problem in "\frac{\partial L}{\partial w}". :arrow_right: https://github.com/larq/larq/pull/228
